### PR TITLE
Add workflow for publishing images to ghcr.io

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Description

This PR adds a new workflow for publishing the API, ingestion server, index worker, and analytics docker images to ghcr.io every time a release is cut. GitHub _should_ just auto-magically allow this repo to publish under `ghcr.io/wordpress`, but we'll see!

This will help eliminate the need to set up Dockerhub credentials as repository secrets. GHCR is also free for open source projects!

This PR is _sort of_ a priority because we have so little space on the catalog EC2 box that I don't want to build the image there, I'd rather pull it from somewhere (but we also don't have easy access to the Dockerhub creds).

See the repositories! :tada: :
- https://ghcr.io/wordpress/openverse-api
- https://ghcr.io/wordpress/openverse-ingestion-server
- https://ghcr.io/wordpress/openverse-index-worker
- https://ghcr.io/wordpress/openverse-analytics

## Testing Instructions

_(will be tested when this PR is opened)_

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by sarayourfriend — APPROVED** [review]
LGTM! This is great!

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+jobs:
+  push_to_registry:
+    name: Push Docker image to GitHub Packages
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+    strategy:
+      matrix:
+        service: [api, ingestion-server, index-worker, analytics]
```
Comment (dhruvkb) [inline .github/workflows/publish.yml:16]: Is this a typo?
```suggestion
        service: [api, ingestion-server, indexer-worker, analytics]
```

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Log in to GitHub Docker Registry
+        uses: docker/login-action@v1
+        with:
+          registry: https://ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and push ${{ matrix.service }} image
```
Comment (dhruvkb) [inline .github/workflows/publish.yml:39]: Not sure if we can use the expressions in the name. Very cool if we can, though!

**Review by dhruvkb — APPROVED** [review]
This is very nice! 

I do have some questions. Does GHCR have a max capacity, and if so, do we need to have some cleanup workflow or does GHCR purge old images on its own?

**Comment by sarayourfriend** [comment]
@dhruvkb it is fully free for OSS repos: https://github.com/features/packages#pricing

File: .github/workflows/publish.yml (hunk 1 above)
Comment (AetherUnbound) [reply .github/workflows/publish.yml:16]: Whoops, yep I'll fix that

File: .github/workflows/publish.yml (hunk 2 above)
Comment (AetherUnbound) [reply .github/workflows/publish.yml:39]: We can! This run is the result of doing so: https://github.com/WordPress/openverse-api/runs/4599659216?check_suite_focus=true

Comment (dhruvkb) [reply .github/workflows/publish.yml:39]: I learn something new about GitHub on a daily basis. Thanks!

**Comment by dhruvkb** [comment]
Referencing #429 as it affects work done in this PR.

**Comment by dhruvkb** [comment]
I made the PR #437 as a proof-of-concept of how #429 would affect the workflow in this repo in a positive way.

**Comment by zackkrida** [comment]
Pinging @rbadillap here so you can take a look at this. It could be relevant to future work on an api deployent pipeline.

**Comment by AetherUnbound** [comment]
Closing in favor of #437

**Pull request closed without merging**